qgis-samgeo-plugin/update_checker.py

```
@@ -0,0 +1,560 @@
+            # Download the zip file
+            self.progress.emit(10, "Downloading plugin from GitHub...")
+
+            def reporthook(block_num, block_size, total_size):
+                # Cooperative cancel: aborts urlretrieve from inside its callback.
+                self._check_canceled()
+                if total_size > 0:
+                    downloaded = block_num * block_size
+                    percent = min(int((downloaded / total_size) * 50), 50)
+                    self.progress.emit(10 + percent, "Downloading...")
+
+            urlretrieve(_require_https(ZIP_URL), zip_path, reporthook)  # nosec B310
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)